salt/modules/virt.py

```diff
@@ -750,6 +760,97 @@ def _get_images_dir():
     return img_dir
 
 
+def _zfs_image_create(vm_name,
+                      disk_filesystem_name,
+                      disk_name,
+                      hostname_property_name,
+                      sparse_volume,
+                      disk_size,
+                      disk_image_name):
+    '''
+    Clones an existing image, or creates a new one.
+
+    When cloning an image, disk_image_name refers to the source
+    of the clone. If not specified, disk_size is used for creating
+    a new zvol, and sparse_volume determines whether to create
+    a thin provisioned volume.
+
+    Set the disk_filesystem_name to the ZFS filesystem in which to
+    create the new zvol.
+
+    The cloned or new volume can have a ZFS property set containing
+    the vm_name. Use hostname_property_name for specifying the key
+    of this ZFS property.
+
+    Example profile for a bhyve VM with two ZFS disks. The first is
+    cloned from the specified image. The second disk is a thin
+    provisioned volume.
+
+    .. code-block:: yaml
+
+        virt:
+          disk:
+            two_zvols:
+              - system:
+                  image: zroot/bhyve/CentOS-7-x86_64-v1@v1.0.5
+                  hostname_property: virt:hostname
+                  disk_filesystem: zroot/bhyve/guests
+              - data:
+                  disk_filesystem: tank/disks
+                  size: 20G
+                  hostname_property: virt:hostname
+                  sparse_volume: True
```

Comment:
This documentation will not be visible to the end user: move this into the ``virt.init`` disks documentation

Reply:
That goes for a lot of documented functions (i.e. disk_profile). The dict keys should be documented at init(), but I feel the rest of the docs are a bit out of place there..